model-optimizer/extensions/ops/split.py

```diff
@@ -44,7 +44,8 @@ def infer(node):
 
         axis = node.in_port(1).data.get_value() if op == 'VariadicSplit' else node.soft_get('axis', None)
         assert axis is not None, '{} `axis` is unknown for node {}'.format(op, name)
-        assert axis.ndim == 0, '{} `axis` should be scalar, but it`s not for node {}'.format(op, name)
+        assert axis.ndim == 0 or len(axis) == 1 and axis.ndim == 1, '{} `axis` should be scalar or tensor with shape ' \
```

Comment:
The condition is wrong. The `ndim` attribute contains rank of the tensor. Also, we should wrap the condition with parentheses to explicitly explain the logic. So I suggest to change the code the following way:

```python
assert axis.ndim == 0 or (axis.ndim == 1 and axis[0] == 1), ...
```

Reply:
But your condition is wrong too. According to `VariadicSplit` specification `axis` value (`axis[0]` in check) value can be from range `-rank(data) .. rank(data)-1`. So with such condition we breaks infer function.

Reply:
Yes, my bad. The condition should be:

```python
assert axis.ndim == 0 or (axis.ndim == 1 and axis.shape[0] == 1), ...
```

Reply:
Condition updated.